docs/src/fees.md

```diff
@@ -0,0 +1,120 @@
+# Fees
+
+Miden transactions pay a fee that is computed and charged automatically by the transaction kernel during the epilogue.
+
+## How fees are computed
+
+- The fee depends on the number of VM cycles the transaction executes and grows logarithmically with that count. The kernel estimates verification cycles by taking log2 of the estimated total cycles, rounding up to the next power of two. The fee amount is then computed as:
+  - `fee_amount = verification_base_fee * ceil(log2(estimated_cycles_rounded_up_to_pow2))`
+- The `verification_base_fee` comes from the current reference block’s fee parameters.
+
+Relevant kernel logic:
+```246:270:crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm
+#! - fee_amount is the computed fee amount of the transaction in the native asset.
+proc.compute_fee
```

Comment:
We usually don't copy code into the docs, mainly because the docs should explain things at a higher-level than code and also because copied code goes stale very fast. So I would remove all copied code here.